spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractPollingEndpoint.java

```diff
@@ -168,6 +169,9 @@ protected void onInit() {
 					this.taskExecutor = new ErrorHandlingTaskExecutor(this.taskExecutor, this.errorHandler);
 				}
 			}
+			if (this.transactionSynchronizationFactory == null) {
+				this.transactionSynchronizationFactory = new PassthroughTransactionSynchronizationFactory();
+			}
```

Comment:
Well, my idea was to fix exactly the TX case. Therefore this `PassthroughTransactionSynchronizationFactory`.
We don't care about transactional poller or not but just use `TransactionSynchronizationManager.isActualTransactionActive()`.
So, that still should be enough for us here as well.
Any other "exotic" advices can access to the message in their after `proceed()` block via the same `ThreadLocal` solution but already as a population in the downstream flow.

Let me know if I have missed something or you need some other explanation.